src/Core.UnitTests/Binding/ServerConnectionTests.cs

```diff
@@ -0,0 +1,149 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Core.Binding;
+using ICredentials = SonarLint.VisualStudio.Core.Binding.ICredentials;
+
+namespace SonarLint.VisualStudio.Core.UnitTests.Binding;
+
+[TestClass]
+public class ServerConnectionTests
+{
+    [TestMethod]
+    public void Ctor_SonarCloudOrg_HasCorrectTypeAndDefaultSettings()
+    {
+        var serverConnection = new ServerConnection("Sonar Cloud Org");
+        
+        serverConnection.Type.Should().Be(ServerConnectionType.SonarCloud);
```

Comment:
Why are we not checking that the id was set? Same feedback for all the other tests